web/pwa/src/features/recorder/FieldEditor.tsx

```diff
@@ -71,6 +71,43 @@ export const DateFieldEditor: FC<FieldEditorProps> = props => {
     </div>
 }
 
+export const MonthFieldEditor: FC<FieldEditorProps> = props => {
+    const {field, value, setValue} = props
+    const expectedLength = 7;
+
+    const [localValue, setLocalValue] = useState(value != null ? (value as Date).toISOString().slice(0, 7) : "")
+
+    const isValidLength = () => localValue.length === expectedLength;
+
+    function isValid(s: string) {
+        const valid = /^(?:19|20|21)\d{2}-[01]\d$/
+        const m = +s.slice(5)
+        return valid.test(s) && 0 < m && m <= 12;
+    }
```

Comment:
Same comment here about dateFns..

Reply:
I've kept the isValid function in place, but replaced the serialisiation with the format function from `date-fns`